Add workflow to update `latest` and `default` branches

KyleFromNVIDIA opened the pull request:
Tested in https://github.com/rapidsai/cudf/pull/18308.

bdice commented on .github/workflows/update-latest-branch.yaml:
```diff
@@ -0,0 +1,91 @@
+              const sha = ref.data.object.sha;
+
+              if (branches.some((b) => b.name == from)) {
+                if (dryRun) {
```
Maybe it would be cleaner to do something like:

```js
// At the top, announce that this is a dry run
if (dryRun) {
  console.log(`This is a dry run. No actions will be taken.`);
}

// For each action, output the logs unconditionally but only act if `!dryRun`
console.log(`Updating branch...`);
if (!dryRun) {
  await github.rest.git.updateRef(...);
}
```

KyleFromNVIDIA replied:
Done: https://github.com/rapidsai/cudf/actions/runs/13952694344/job/39057321557?pr=18308

bdice commented on .github/workflows/update-latest-branch.yaml:
```diff
@@ -0,0 +1,91 @@
+              }
+              return prev;
+            }, null).name;
+            await tryUpdateBranch(latestBranch, "latest");
```
With the current structure, `tryUpdateBranch` needs three pieces of information: the branch that triggered this action (current branch), the "to" branch (e.g. `branch-25.06`), and the "from" branch (`latest`).

The implementation is basically: `if ${currentBranch} == ${to}, then update the head of ${from} to match the head of ${to}`. Is that right?

The naming and logic feel backwards to me, if I am reading that correctly.

Maybe we can choose a different set of names, like `if ${currentBranch} == ${source}, then update the head of ${target} to match the head of ${source}`. Maybe I'm just reading it funny.

I do think we could pull out `${currentBranch}` so that it isn't needed inside of `tryUpdateBranch` at all. That should only need two pieces of information (to/from) and the check for `if ${currentBranch} == ${to}` should happen outside of the update function. Maybe something like this:

```js
// Do the update. No conditions.
const updateBranch = async (to, from) => { ... }

// Check whether to update, and maybe update.
const tryUpdateBranch = async(current, to, from) => { ... }

// Wrap logic for getting the latest branch in a function
const getLatestBranch = (branches) => { ... }
```

bdice commented on .github/workflows/update-latest-branch.yaml:
```diff
@@ -0,0 +1,91 @@
+                if (dryRun) {
+                  console.log(`Would have created branch "${from}" to point to branch "${to}", SHA ${sha}`);
+                } else {
+                  console.log(`Creating branch "${from}" to point to branch "${to}", SHA ${sha}`);
```
I'm not sure if I want to let this bot create branches. If the bot creates a branch, it won't have the right protections. Perhaps we should require a manual step from ops to create the branch first, and set the right protections.

KyleFromNVIDIA commented:
@bdice I've taken your feedback, removed an unnecessary API call, and added an `all` input to update all branches instead of just the one that got pushed. You can see the results [here](https://github.com/rapidsai/cudf/actions/runs/13953603941/job/39059181731?pr=18308).

bdice reviewed (approved):
Love it! This is really nice.

bdice commented:
@KyleFromNVIDIA Following up on this -- is there an issue tracking how this will be rolled out / used in RAPIDS? I don't see one cross-linked here.

KyleFromNVIDIA commented:
Oh, but we also realized that the token in the workflow won't have permission to push to protected branches, and we need an app or something.
